openeo-geotrellis/src/main/scala/org/openeo/geotrellis/file/FileRDDFactory.scala

```diff
@@ -27,7 +26,7 @@ import scala.collection.JavaConverters._
 class FileRDDFactory(openSearchCollectionId: String, openSearchLinkTitles: util.List[String],attributeValues: util.Map[String, Any] = util.Collections.emptyMap(),correlationId: String = "") {
 
   private val maxSpatialResolution = CellSize(10, 10)
-  protected val openSearch: OpenSearch = OpenSearch(new URL("http://oscars-01.vgt.vito.be:8080"))
+  protected val openSearch: OscarsOpenSearch = OscarsOpenSearch(new URL("http://oscars-01.vgt.vito.be:8080"))
```

Comment:
The type here should remain OpenSearch, not a specific subclass, so we can ensure that we working against the abstract base class.

Reply:
Only problem to change this is that the method signatures differ because we need to pass processingLevel in creo. Maybe that can be a constructor parameter?

The getCollections() return type is also not the same because the output is different in oscars and creo. Maybe we don't need that function in creo, because the one in oscars is only used to fetch the extent for the metadata and that information isn't available in the creo collections response.

Reply:
One solution: pass parameters in a more generic way, like a map.
Other option: also add processinglevel to oscars implementation and not use it?

Reply:
Maybe I'm misunderstanding, but FileRDDFactory should also be usable against creo open search in some way, right?

Reply:
Looks better now!